### ISSUE TITLE
Prompt for Shift Code platform choice

### ISSUE THREAD

**Issue by VxJasonxV** [matt1484/bl3_auto_vip#32]
If a Shift Code applies to multiple platforms, the user needs to be able to specify which platform to redeem it for. As currently implemented, had I not already manually redeemed shift codes prior to the use of this app, they would have redeemed for the XB1 which is worthless to me as I play via Epic/PC.

When a new one comes out, if I use the app, it will be redeemed for the XB1 which will no doubt annoy me immensely.

This app should either prompt for platform and store (like credentials), or simply print availability and *not* automatically redeem.

**Comment by squatto** [comment]
@VxJasonxV I'm just curious, but why do you care if it's redeemed for a platform that you don't actively play on? It takes a fraction of a second, doesn't affect anything, and you'll never even know. The only potential outcome is that if some day you decide to play on XB1, you'll have a bunch of golden keys to use. I'm not sure I understand where the problem is?

**Comment by squatto** [comment]
One of the first things that this program does is retrieve your SHIFT profile and pull out the platforms that you have linked to it. It will only attempt to redeem SHIFT codes on those platforms. In other words, if you don't have your XBOX Live account linked to your SHIFT account, it won't redeem XB1 codes.

For example, my SHIFT account has these links:

![image](https://user-images.githubusercontent.com/748444/69278050-bf461180-0b9e-11ea-9409-6481ea6b65a5.png)

The program pulls this from my SHIFT profile: (the same 9 in the image)

```json5
{
  //...
  "platforms": [
    "steam",
    "twitch",
    "my2k",
    "epic",
    "xboxlive",
    "googleplay",
    "psn",
    "facebook",
    "twitter"
  ],
  //...
}
```

**Comment by VxJasonxV** [comment]
I care because you can only redeem a shift code on one single platform. You can’t redeem the same code on Xbox and PS4 and Epic and Steam and …
(I actually accidentally redeemed a BL3 code for Steam, which I don’t intend to buy it there as I’ve been playing on Epic.)

My understanding is that there are one use per account shift codes, and platform specific shift codes (historically, at least).

It sounds like the solution is unlinking those accounts, but I’m unsure of any other ramifications in doing that.

I did play Pre-Sequel on Steam, am playing BL3 on Epic. Played 1+2 on XB360.

**Comment by squatto** [comment]
I’ve redeemed codes on multiple platforms a ton of times. Where did you see that you can only redeem it on one platform? The universal codes can be redeemed on any supported platform, as far as I’m aware.

**Comment by VxJasonxV** [comment]
The issue here was that I had no shift code to test with, and never saved program output. It was only after I'd redeemed a code that I had this idea, and IIRC there have been no codes between when I filed this issue and last week.

I don't know if this is a program version, shift code source, or whatever issue, but running bl3 auto today didn't redeem the new shift code. However, I did redeem it directly through the shift web interface, and was able to do it for every single platform, good enough proof for me that this issue is invalid.

Sorry for the noise.
